docs/src/main/sphinx/security/opa-access-control.md

```diff
@@ -304,6 +308,134 @@ column.
+masking.
+
+An OPA policy supporting batch column masking must return a list of objects,
+each containing:
+- `viewExpression`: an object made of:
+    - `expression`: the expression to apply to the column, as a string
+    - `identity` (optional): the identity to evaluate the expression as, as a
+      string
+- `index`: a reference the index of the column in the request to which this mask
+  applies
+
+For example, a policy configuring batch column masking may be defined by the
+following script:
+
+```text
+package trino
+import future.keywords.in
+import future.keywords.if
+import future.keywords.contains
+
+default allow := true
+
+batchColumnMasks contains {
+  "index": i,
+  "viewExpression": {
+    "expression": "NULL"
+  }
+} if {
+  some i
+  column_resource := input.action.filterResources[i]
+  column_resource.catalogName == "sample_catalog"
+  column_resource.schemaName == "sample_schema"
+  column_resource.tableName == "restricted_table"
+  column_resource.columnName == "user_phone"
+}
+
+
+batchColumnMasks contains {
+  "index": i,
+  "viewExpression": {
+    "expression": "'****' || substring(user_name, -3)",
+    "identity": "admin"
+  }
+} if {
+  some i
+  column_resource := input.action.filterResources[i]
+  column_resource.catalogName == "sample_catalog"
+  column_resource.schemaName == "sample_schema"
+  column_resource.tableName == "restricted_table"
+  column_resource.columnName == "user_name"
+}
+```
+
+A batch column masking request is similar to the following example:
+
+```json
+{
+    "context": {
+        "identity": {
+            "user": "foo",
+            "groups": ["some-group"]
+        },
+        "softwareStack": {
+            "trinoVersion": "434"
+        }
+    },
+    "action": {
+        "operation": "GetColumnMask",
+        "filterResources": [
+            {
+                "column": {
+                    "catalogName": "sample_catalog",
+                    "schemaName": "sample_schema",
+                    "tableName": "restricted_table",
+                    "columnName": "user_phone",
+                    "columnType": "VARCHAR"
+                }
+            },
+            {
+                "column": {
+                    "catalogName": "sample_catalog",
+                    "schemaName": "sample_schema",
+                    "tableName": "restricted_table",
+                    "columnName": "user_name",
+                    "columnType": "VARCHAR"
+                }
+            }
+        ]
+    }
+}
+```
+
+For which the OPA response would be:
+
+```json
+[
+    {
+        "index": 0,
+        "viewExpression": {
+            "expression": "NULL"
+        }
+    },
+    {
+        "index": 1,
+        "viewExpression": {
+            "expression": "'****' || substring(user_name, -3)",
+            "identity": "admin"
+        }
+    }
+]
+```
+
+Making Trino apply expression `NULL` to the column at index 0 (`user_phone`)
```

Comment:
This is not a sentence.. what are you trying to say?

Reply:
This was trying to explain the effects of the the returned column masks. Removed.